src/style.css

```diff
@@ -0,0 +1,113 @@
+body {
+  --logo-big-size: 19px;
+  --logo-small-size: 11px;
+  --big--search-size: 70px;
+  --small--search-size: 42px;
+  --search-field-height: 19px;
+  --text-shadow: 0 4px 4px rgba(0,0,0,0.25);
+}
+
+.search--box {
+  border-radius: 4px;
+  border: 1px solid #e1e7ed;
+  background: #fff;
+  box-shadow: 0 1px 8px 0 rgba(61, 78, 97, 0.1);
+  height: var(--big--search-size);
+  display: flex;
+  box-sizing: border-box;
+  align-items: center;
+  margin-top: 20px;
+  padding-left: 26px;
+}
+
+.search--box:hover {
+  box-shadow: 0 3px 8px 0 rgba(61, 78, 97, 0.2);
+}
+
+.search--box--small {
+  height: var(--small--search-size);
+  padding-left: 13px;
+}
+
+.search--bar {
+  font-family: "Avenir", serif;
+  color: #3d4e61;
+  font-size: 16px;
+  box-sizing: border-box;
+  border: none;
+  margin-left: 17px;
+  height: var(--search-field-height);
+  width: 100vw;
+}
+
+.search--bar--small {
+  font-size: 14px;
+  margin-left: 9px;
+}
+
+.search--bar::placeholder {
+  color: #3d4e61;
+}
+
+.search--bar--small::placeholder {
+  text-shadow: var(--text-shadow);
+}
+
+.search--bar:focus {
+  outline: none;
+  font-weight: 900;
+  /*Because of gradient applying transparent when it's focused*/
+  background: transparent;
+  border-radius: 3px;
+}
+
+/*Reset havy font when focused for placeholder*/
+.search--bar:focus::placeholder {
+  font-weight: normal;
+}
+
+/*Remove placeholders' text shadow of big search when focused
+(shadow style sets to whole input when focused including placeholder) */
+.search--bar--big:focus::placeholder {
+  text-shadow: none;
+}
+
+/*Remove box shadow of small search when focused */
+.search--bar--small:focus {
+  box-shadow: none;
+}
+
+/*Set up background to container when forms' child (input) focused*/
+.search--box:focus-within {
+  background: linear-gradient(180deg, #fff 0%, #f6f6f7 100%) !important;
+}
+
+/*Add shadow to input text in big container when input focused*/
+.search--bar--big:focus.search--bar--big {
+  text-shadow: var(--text-shadow);
+}
```

Comment:
Can not agree with this comment. I implement figma`s design in all details, and here is description for selectors:
1. This block required to set font-color for both placeholders
```
.search--bar::placeholder {
  color: #3d4e61;
}
```
2.  This adds shadow only into small-bars placeholder ass it's designed in figma (without this rule there will be no shadow in small placeholder and this block is only way to implement this feature)
```
.search--bar--small::placeholder {
  text-shadow: var(--text-shadow);
}
```

3. Default focus properties
```
.search--bar:focus {
  outline: none;
  font-weight: 900;
  /*Because of gradient applying transparent when it's focused*/
  background: transparent;
  border-radius: 3px;
}
```

4. In block 3, when bar focused, we sets up `font-weight: 900;`. As noted in checklist's point 11: "Placeholder shoudn't change style on focus". So this block resets bold font weight for placeholder even if bar focused.
```
.search--bar:focus::placeholder {
  font-weight: normal;
}
```

5. This block adds shadow to input text in big search when it is focused. Separate block because shadow needed only in big search. (I remove `.search--bar--big` from selector cause it is really no need in it)
```
.search--bar--big:focus {
  text-shadow: var(--text-shadow);
}
```

6. Likewise point 4 in this list, this block is required to override shadow that sets up in point 5 of this list (for big search every time it's focused), required because of point 11 in checklist
```
.search--bar--big:focus::placeholder {
  text-shadow: none;
}
```

7. This block is required for shadow removing in small search when focused (big should have shadow even being focused so we need separate selector)
```
.search--bar--small:focus {
  box-shadow: none;
}
```

8. This block required because we shold set background to search--box when focused it's child (search--bar). For these purpose we have `:focus-within`
.search--box:focus-within {
  background: linear-gradient(180deg, #fff 0%, #f6f6f7 100%) !important;
}

If u still think I should decrease selectors amount - provide some instructions, cause all of declared selectors are required in my implementation